containers/base/Dockerfile

```diff
@@ -1,252 +1,293 @@
+    rm -rf /usr/share/i18n/locales/* && \
+    cd /
+
+ENV LC_ALL en_US.UTF-8
+ENV LANG en_US.UTF-8
+
+ADD config/ipython.py /etc/ipython/ipython_config.py
+ADD config/nbconvert.py /etc/jupyter/jupyter_notebook_config.py
+
+# Directory "py" may be empty and in that case it will git clone pydatalab from repo
+ADD pydatalab /datalab/lib/pydatalab
+ADD nbconvert /datalab/nbconvert
+
+# Do IPython configuration and install build artifacts
+# Then link stuff needed for nbconvert to a location where Jinja will find it.
+# I'd prefer to just use absolute path in Jinja imports but those don't work.
+RUN ipython profile create default && \
+    jupyter notebook --generate-config
+
+RUN if [ -d /datalab/lib/pydatalab/.git ]; then \
+        echo "use local lib"; \
+      else \
+        git clone https://github.com/googledatalab/pydatalab.git /datalab/lib/pydatalab; \
+      fi
+RUN cd /datalab/lib/pydatalab && \
+    /tools/node/bin/npm install -g typescript && \
+    tsc --module amd --noImplicitAny --outdir datalab/notebook/static datalab/notebook/static/*.ts && \
+    tsc --module amd --noImplicitAny --outdir google/datalab/notebook/static google/datalab/notebook/static/*.ts && \
+    /tools/node/bin/npm uninstall -g typescript
+RUN cd /datalab/lib/pydatalab && \
+    pip install --upgrade-strategy only-if-needed --no-cache-dir . && \
+    pip install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/image_classification/. && \
+    pip install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
+    pip3 install --upgrade-strategy only-if-needed --no-cache-dir . && \
+    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/image_classification/. && \
+    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
+    jupyter nbextension install --py datalab.notebook && \
+    jupyter nbextension install --py google.datalab.notebook && \
+    jupyter nbextension enable --py widgetsnbextension && \
+    rm datalab/notebook/static/*.js google/datalab/notebook/static/*.js && \
+    mkdir -p /datalab/nbconvert && \
+    cp -R /usr/local/share/jupyter/nbextensions/gcpdatalab/* /datalab/nbconvert && \
+    ln -s /usr/local/lib/python2.7/dist-packages/notebook/static/custom/custom.css /datalab/nbconvert/custom.css && \
+    cd /
+
+## Use Debian unstable via pinning -- new style via APT::Default-Release
+RUN echo "deb http://http.debian.net/debian sid main" > /etc/apt/sources.list.d/debian-unstable.list \
+	&& echo 'APT::Default-Release "testing";' > /etc/apt/apt.conf.d/default
+
+ENV R_BASE_VERSION 3.4.3
+
+## Now install R and littler, and create a link for littler in /usr/local/bin
+## Also set a default CRAN repo, and make sure littler knows about it too
+RUN apt-get update \
+	&& apt-get install -t unstable -y --no-install-recommends \
+		libssh2-1-dev \
+		libcurl4-openssl-dev \
+		libssl-dev \
+		littler \

```

Comment:
littler is GPL licensed.  It needs to be included in the list on line 46 so that we distribute the source code.

Same with r-base, r-base-dev, and r-recommended, I think.